connectors/src/connectors/notion/temporal/activities.ts

```diff
@@ -829,19 +837,15 @@ export async function garbageCollect({
   let stillAccessiblePagesCount = 0;
   let stillAccessibleDatabasesCount = 0;
 
-  // Find the resources not seen in the GC run
-  const resourcesToCheck = await findResourcesNotSeenInGarbageCollectionRun(
-    connector.id
-  );
-
-  for (const [i, x] of resourcesToCheck.entries()) {
+  batch.forEach(async (x, i) => {
```

Comment:
nit: why the forEach with an async callback vs a simple for loop ?

Reply:
Also you are neither not awaiting the promise and the errors aren't handled right ?

Don't we want to control concurrency instead of just creating all the promises for the whole batch ?

Reply:
> nit: why the forEach with an async callback vs a simple for loop ?

Typescript would complain that I may be accessing something that turn into undefined (if I exceed the number of element).

> Also you are neither not awaiting the promise

That's a mistake.

> the errors aren't handled right ?

I'm confused by that part, let's discuss IRL.

Reply:
> Don't we want to control concurrency

Isn't what PQueue does ?